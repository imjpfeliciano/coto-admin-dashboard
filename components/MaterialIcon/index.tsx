import styled from "styled-components";
import theme from "../../utils/theme";

interface IconProps {
  iconName: string;
  onClick?: () => void;
  color?: string;
}

const MaterialIcon: React.FC<IconProps> = ({ iconName, onClick, color }) => {
  // FIXME: hover colors
  const classColor = color === "white" ? "text-white dark:text-black group-hover:text-gray-400" : `text-${color}-500  dark:text-${color}-400 group-hover:text-${color}-900`;
  return (
    <span className={`material-icons w-6 h-6 ${classColor} transition duration-75  dark:group-hover:text-white`}>
      {iconName}
    </span>
  );
}
export default MaterialIcon;
