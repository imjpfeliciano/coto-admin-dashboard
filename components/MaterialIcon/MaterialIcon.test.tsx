import { render } from '@testing-library/react'
import MaterialIcon from '.'

describe('Material Icon', () => {
  it('should render', () => {
    const { container } = render(<MaterialIcon iconName='add' />)
    expect(container).toMatchSnapshot()
  })

  it('should execute onClick function', () => {
    const onClick = jest.fn()
    const { getByText } = render(
      <MaterialIcon iconName='add' onClick={onClick} />
    )
    getByText('add').click()
    expect(onClick).toHaveBeenCalled()
  })
})
