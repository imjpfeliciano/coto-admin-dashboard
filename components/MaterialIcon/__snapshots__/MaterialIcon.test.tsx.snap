// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Material Icon should render 1`] = `
<div>
  <span
    class="MaterialIcon__IconContainer-sc-1s3bnun-0 fVqaYP material-icons"
  >
    add
  </span>
</div>
`;
